### PR TITLE
Fix/remove character restriction in name

### LINKED ISSUE #225: Umlaute should be valid characters in usernames
Currently usernames cannot contain any [umlaute](https://en.wikipedia.org/wiki/Umlaut_(diacritic)), e.g. `öüä` or the [ß](https://en.wikipedia.org/wiki/%C3%9F). 

It can be reproduced by either making a new account or changing the name of an existing one.

Should work with umlaute and other normal characters.

<details>
<summary><b>Example of the problem</b></summary>
image
</details>

### PR DESCRIPTION
**Pull request by FredrikMorstad**
## 🌐 Fix only Norwegian latters allowed in names
Resolved #225
Now all alphabets are valid characters in name all name validation fields
#### Input field overflowing parent div 🐛 
Currently the registration looks like this
Screenshot 2023-12-22 at 16 14 23
Now:
Screenshot 2023-12-22 at 16 15 10

## Update with `-` and `'`
Now the regex only allow `-` and `'` when between characters so `---` or `   ` is not a valid name 
https://github.com/td-org-uit-no/tdctl-frontend/assets/43537864/fa1c6969-e696-40f3-bbc7-3afe0dbabbeb

### CODE REVIEW

File: src/utils/validators.ts [hunk 1]
```diff
@@ -55,11 +55,18 @@ export const maxParticipantsValidator = (
 };
 
 export const nameValidator = (name: string): string[] | undefined => {
-  const nameReg = /^[a-zæøåA-ZÆØÅ ]+$/; //! note the last space
+  // \p{L} -> letter in any language
+  // \p{Zz} -> allows space
+  // 1,30 is the valid range
+  // must be an range or any name over one char gets an error
+  const nameReg = /^[\p{L}\p{Zs}]{1,30}$/u;
```
Comment (otytlandsvik) [inline src/utils/validators.ts:62]: Should also allow `-`, at least in the last name (ex: Marius Møller-Hansen should be valid).

File: src/utils/validators.ts [hunk 2]
```diff
@@ -55,11 +55,18 @@ export const maxParticipantsValidator = (
...
 
   if (name.length === 0) {
     return ['Navn er påkrevd'];
   }
+  if (name.length > 30) {
+    return ['Navnet er for langt, maks 30 bokstaver'];
+  }
   if (!new RegExp(nameReg).test(name)) {
     return ['Kun bokstavene A-Å'];
```
Comment (otytlandsvik) [inline src/utils/validators.ts:71]: Could maybe give a different message here, since we allow more than A-Å now

**Review by otytlandsvik — APPROVED** [review]
I think you should add `-` to the regex. Otherwise, LGTM :rocket:

File: src/utils/validators.ts (hunk 2 above)
Comment (FredrikMorstad) [reply src/utils/validators.ts:71]: whoops, forgot about that one

File: src/utils/validators.ts (hunk 1 above)
Comment (FredrikMorstad) [reply src/utils/validators.ts:62]: Well, I left any of these characters out as it will allow for name such as `---`

Comment (FredrikMorstad) [reply src/utils/validators.ts:62]: fixed now

**Pull request merged**